include/ignition/math/Helpers.hh

```diff
@@ -727,6 +729,43 @@ namespace ignition
       }
     }
 
+    /// \brief Convert a std::chrono::steady_clock::time_point to a seconds and
+    /// nanoseconds pair.
+    /// \param[in] _time The time point to convert.
+    /// \return A pair where the first element is the number of seconds and
+    /// the second is the number of nanoseconds.
+    inline std::pair<int64_t, int64_t> timePointToSecNsec(
+        const std::chrono::system_clock::time_point &_time)
+    {
+      auto now_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
+        _time.time_since_epoch());
+      auto now_s = std::chrono::duration_cast<std::chrono::seconds>(
+        _time.time_since_epoch());
+      int64_t seconds = std::chrono::duration_cast<std::chrono::seconds>(
+        _time.time_since_epoch()).count();
+      int64_t nanoseconds = std::chrono::duration_cast
+        <std::chrono::nanoseconds>(now_ns - now_s).count();
+      return {seconds, nanoseconds};
+    }
+
+    /// \brief Convert to a seconds and nanoseconds to
+    /// std::chrono::steady_clock::time_point.
+    /// \param[in] _sec The seconds to convert.
+    /// \param[in] _nanosec The nanoseconds to convert.
+    /// \return A std::chrono::system_clock::time_poin based on the number of
+    /// seconds and
+    /// the number of nanoseconds.
+    inline std::chrono::system_clock::time_point secNsecToTimePoint(
+        const uint64_t &_sec, const uint64_t &_nanosec)
+    {
+      auto duration = std::chrono::seconds(_sec) + std::chrono::nanoseconds(
+        _nanosec);
+      std::chrono::system_clock::time_point result =
+        std::chrono::system_clock::from_time_t(0);
+      result = result + duration;
```

Comment:
Small nit: could condense with `result += duration`, I'll leave it up to you though, same with all below additions like this made in the tests

Reply:
This line is failing in OSX https://build.osrfoundation.org/job/ignition_math-ci-pr_any-homebrew-amd64/666/consoleFull#1943795879aa60f765-a60a-427d-900c-dc128ab22287 even if you use your suggestion. Any thoughts?

Reply:
sorry for making suggestions that don't actually work; I thought it would be cleaner. I'll take a look with my mac later today

Reply:
I looked into this a bit:  Seems like OSX stores `chrono` to only a microsecond precision, whereas Ubuntu goes to nanosecond precision.  It seems like all platforms have different precisions.  See https://github.com/eddelbuettel/nanotime/issues/74 and https://stackoverflow.com/questions/55120594/different-behaviour-of-system-clock-on-windows-and-linux